### PR TITLE
feat: add support for network_device MTU

### PR DESCRIPTION
**Pull request by abdo-farag**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

### CODE REVIEW

**Comment by bpg** [comment]
Hey @abdo-farag thanks for contributing this! I'll take a closer look at the code later this week.

The disk problem you've mentioned in the comment is interesting though, don't remember seeing this before. I don't think we can keep that if condition commented, as it may break some other use cases... 
Would you mind sharing your VM template so I can take a closer look?

**Comment by bpg** [comment]
Thanks for the details, I've created a separate ticket for this error https://github.com/bpg/terraform-provider-proxmox/issues/179, a fix is on the way.

File: proxmoxtf/resource_virtual_environment_container.go [hunk 1]
```diff
@@ -1396,12 +1420,18 @@ func resourceVirtualEnvironmentContainerRead(ctx context.Context, d *schema.Reso
...
 
+	/*
+		I can't use disk datastore otherthan local-lvm and it cause the error
+		"Error: Invalid address to set: []string{"datastore_id"}"
+	*/
+	// } else if len(currentDisk) > 0 ||
+	// 	disk[mkResourceVirtualEnvironmentContainerDiskDatastoreID] != dvResourceVirtualEnvironmentContainerDiskDatastoreID {
+	// 	err := d.Set(mkResourceVirtualEnvironmentContainerDiskDatastoreID, []interface{}{disk})
+	// 	diags = append(diags, diag.FromErr(err)...)
+	// }
```
Comment (bpg-dev) [inline proxmoxtf/resource_virtual_environment_container.go]: @abdo-farag I pushed the fix, could you rebase and uncomment this block to see if it works for you?

Comment (abdo-farag) [reply proxmoxtf/resource_virtual_environment_container.go]: @bpg It works!, Thanks

Comment (bpg-dev) [reply proxmoxtf/resource_virtual_environment_container.go]: > @bpg It works!, Thanks

Great! Could you please put the SSD work in a separate PR, and we'll focus here on MTU?
Thanks!

Comment (abdo-farag) [reply proxmoxtf/resource_virtual_environment_container.go]: @bpg I have created new branch for ssd feature is it better so? I don't know how to split them :)

**Review by bpg — CHANGES_REQUESTED** [review]
Hey @abdo-farag! The code looks great, it seems to be working fine in my tests as well. I've pushed a small commit to add `mtu` arguments to the examples.

Could you please add the new MTU parameters to the docs? [virtual_environment_container.md](https://github.com/bpg/terraform-provider-proxmox/blob/main/docs/resources/virtual_environment_container.md) and [virtual_environment_container.md](https://github.com/bpg/terraform-provider-proxmox/blob/main/docs/resources/virtual_environment_container.md) to complete the PR? It is also worth to mention that MTU can't be greater than the bridge MTU of the PVE.

**Review by bpg — APPROVED** [review]
Hey @abdo-farag, I've updated the docs. Also changed default MTU from 1500 -> 0 as I don't think 1500 can be universally applied. The VM / LXC MTU can't be larger than the bridge MTU, and that one potentially can be lower than 1500 in some cases.
Other than that - 🚢 !

**Pull request merged**